python/lance/tests/test_schema_evolution.py

```diff
@@ -30,3 +32,34 @@ def test_write_versioned_dataset(tmp_path: Path):
     actual_df = new_dataset.to_table().to_pandas()
     expected_df = pd.DataFrame({"a": [1, 10], "b": [2, 20], "c": ["a0", "a1"]})
     pd.testing.assert_frame_equal(expected_df, actual_df)
+
+
+def test_add_column_with_literal(tmp_path: Path):
+    table = pa.Table.from_pylist([{"a": i} for i in range(10)])
+
+    base_dir = tmp_path / "test"
+    lance.write_dataset(table, base_dir)
+    dataset = lance.dataset(base_dir)
+    new_dataset = dataset.append_column(pa.field("b", pa.float64()), pc.scalar(0.5))
+
+    assert new_dataset.version["version"] == 2
+    actual_df = new_dataset.to_table().to_pandas()
+    expected_df = table.to_pandas()
+    expected_df["b"] = 0.5
+    pd.testing.assert_frame_equal(actual_df, expected_df)
+
+
+def test_add_column_with_compute(tmp_path: Path):
```

Comment:
There's not test yet for array (pre computed whole column) or function?

Reply:
pre-computed array has not been supported yet.

function test is in this file (from the previous PR)

https://github.com/eto-ai/lance/pull/324/files/1dd095b23e1a7a15cb703a6d6b2b308d420e0198#diff-ed974e93ecd35d82f64c13c4a70f71f6ee757911825654a2856b9d06cf6164c9R24